app/src/main/java/kth/jjve/xfran/repositories/WorkoutRepository.java

```diff
@@ -25,17 +32,75 @@ public static WorkoutRepository getInstance(){
     }
 
     // Pretend to get data from a webservice or cage or online source
-    public MutableLiveData<List<Workout>> getWorkouts(){
+    public MutableLiveData<List<Workout>> getWorkouts() throws IOException {
         setWorkouts(); //Todo: normally retrieve data from webservice or anything here
         MutableLiveData<List<Workout>> data = new MutableLiveData<>();
         data.setValue(dataSet);
         return data;
     }
 
-    private void setWorkouts(){
+    private void setWorkouts() throws IOException {
         // This mimics what the database would normally do
-        dataSet.add(new Workout("Fran", "21-12"));
-        dataSet.add(new Workout("DT", "10-12"));
-        dataSet.add(new Workout("Long jump", "10-12"));
+        // For now it reads a json file with workout info
+        //open json file
+        InputStream workoutData  = AppCtx.getContext().getResources().openRawResource(R.raw.workouts_data);
+        //initialize reader
+        JsonReader reader = new JsonReader(new InputStreamReader(workoutData, "UTF-8"));
+        try {
+            parseWorkouts(reader);
+        } finally {
+            reader.close();
+        }
+    }
+
+    public void parseWorkouts(JsonReader reader) throws IOException {
+        //read json array, loop through workouts on dataSet and read each of them
+        reader.beginArray();
+        while (reader.hasNext()) {
+            dataSet.add(readWorkout(reader));
+        }
+        reader.endArray();
+    }
+
+    public Workout readWorkout(JsonReader reader) throws IOException {
+        //initialize workout parameters
+        String title = null;
+        String type = null;
+        ArrayList<String> exercises = null;
+        //read workout info
+        reader.beginObject();
+        while (reader.hasNext()) {
+            String name = reader.nextName();
+            if (name.equals("name")) {
+                title = reader.nextString();
+            } else if (name.equals("description")) {
+                type = reader.nextString();
+            } else if (name.equals("excercises")) {
+                //exercise is an array therefore requires a reader
+                exercises = readExercises(reader);
+            } else {
+                reader.skipValue();
+            }
+        }
+        reader.endObject();
+        //create workout object from info read
+        Workout workout = new Workout(title);
+        workout.setExercises(exercises);
+        workout.setType(type);
+        //Log.i(LOG_TAG, "workout: "+workout);
+        return workout;
+    }
+
+    public ArrayList<String> readExercises(JsonReader reader) throws IOException {
+        //exercises for each workout obj are stored in an ArrayList
+        ArrayList<String> exercises = new ArrayList<String>();
+        //loop through exercises and read each of them
```

Comment:
this is a comment to answer a question that is also answered by the code -> not usefull :)